compose.yml

```diff
@@ -0,0 +1,47 @@
+version: '3.1'
+
+services:
+  redis:
+    image: redis
+    ports: ['6379:6379']
+    healthcheck:
+      test: ["CMD", "redis-cli","ping"]
+
+  postgres:
+    image: postgres:15
+    ports: ['5432:5432']
+    restart: always
+    environment:
+      POSTGRES_USER: app
+      POSTGRES_DB: app_development
+      POSTGRES_PASSWORD: secret
+    volumes:
+    - pgdata:/var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "app", "-d", "app_development"]
+      interval: 1s
+      retries: 5
+
+  server: &server
+    depends_on: ["redis", "postgres"]
```

Comment:
I see that server depends on redis, but when I tried this locally I get an error about redis when I try submitting a message. Console shows:
```
server-1    | Redis::CannotConnectError (Address not available - connect(2) for [::1]:6379):
```

I just started up docker compose and hit `localhost:3000` and that first page load and registering a user worked properly, but it was when I submitted a chat message that I got the redis error. It seems like everything started up properly:

```
+] Running 4/4
 ✔ Container hostedgpt-redis-1     Created                                                                                                           0.0s 
 ✔ Container hostedgpt-postgres-1  Running                                                                                                           0.0s 
 ✔ Container hostedgpt-worker-1    Recreated                                                                                                         0.1s 
 ✔ Container hostedgpt-server-1    Recreated                                                                                                         0.1s 
Attaching to postgres-1, redis-1, server-1, worker-1
redis-1     | 1:C 14 Feb 2024 21:36:12.999 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-1     | 1:C 14 Feb 2024 21:36:12.999 * Redis version=7.2.4, bits=64, commit=00000000, modified=0, pid=1, just started
redis-1     | 1:C 14 Feb 2024 21:36:12.999 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-1     | 1:M 14 Feb 2024 21:36:12.999 * monotonic clock: POSIX clock_gettime
redis-1     | 1:M 14 Feb 2024 21:36:13.000 * Running mode=standalone, port=6379.
redis-1     | 1:M 14 Feb 2024 21:36:13.000 * Server initialized
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * Loading RDB produced by version 7.2.4
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * RDB age 167498 seconds
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * RDB memory usage when created 0.88 Mb
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * Done loading RDB, keys loaded: 0, keys expired: 0.
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * DB loaded from disk: 0.001 seconds
redis-1     | 1:M 14 Feb 2024 21:36:13.001 * Ready to accept connections tcp
worker-1    | Skipping db:prepare
server-1    | Running db:prepare
worker-1    | [SolidQueue] Starting Dispatcher(pid=14, hostname=a6a379bf5328, metadata={:polling_interval=>1, :batch_size=>500})
worker-1    | [SolidQueue] Starting Worker(pid=16, hostname=a6a379bf5328, metadata={:polling_interval=>0.1, :queues=>"*", :thread_pool_size=>5})
server-1    | => Booting Puma
server-1    | => Rails 7.1.3 application starting in development 
server-1    | => Run `bin/rails server --help` for more startup options
server-1    | Puma starting in single mode...
server-1    | * Puma version: 6.4.0 (ruby 3.2.3-p157) ("The Eagle of Durango")
server-1    | *  Min threads: 5
server-1    | *  Max threads: 5
server-1    | *  Environment: development
server-1    | *          PID: 1
server-1    | * Listening on http://0.0.0.0:3000
server-1    | Use Ctrl-C to stop
```